libc/src/stdlib/qsort_util.h

```diff
@@ -27,11 +27,46 @@
 namespace LIBC_NAMESPACE_DECL {
 namespace internal {
 
+template <typename A, typename F> void sort_inst(A &array, const F &is_less) {
```

Comment:
Naming the comparison `is_less` is a bit confusing since it's not a binary comparison. The comparison function returns a positive, negative, or zero integer if the first element is greater than, less than, or equal to.

Reply:
Uhh, it is a binary comparison .. the name is the same used in the code it is ported from. Take a look at how it is used in the code.

Reply:
The name `is_less` implies it returns a boolean, e.g. "a is less than b". Using a name like `compare` makes it more clear this a comparison function that returns a direction, similar to strcmp. A more clear name makes it easier for everyone to understand the code, you included. See my comment in `quick_sort.h`.

Reply:
@michaelrj-google I do believe it returns a boolean, see [`libc/src/stdlib/qsort.cpp`](https://github.com/llvm/llvm-project/pull/120450/files/efb98579a7f56c721402d526086fa24a9cbc3110#diff-26ff238bd5dfa4e151c74fb30555d8b07621bbf887766762dbf4774f1c9f5c2f) where `internal::quicksort` is instantiated with `compare(a, b) < 0`.